sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/_metrics_handler/_code_metric_handler.py

```diff
@@ -0,0 +1,108 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import logging
+import tqdm
+from concurrent.futures.thread import ThreadPoolExecutor
+
+from numpy import NaN
+
+from .._metric_handler import MetricHandler
+from ..metrics._custom_metric import CodeMetric
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CodeMetricHandler(MetricHandler):
+    def __init__(
+            self,
+            task_type,
+            prediction_data,
+            test_data,
+            truth_data=None,
+            prediction_data_column_name=None,
+            ground_truth_column_name=None,
+            metrics_mapping=None,
+            metrics=None,
+            type_to_kwargs=None,
+    ):
+
+        super().__init__(
+            task_type=task_type,
+            prediction_data=prediction_data,
+            test_data=test_data,
+            truth_data=truth_data,
+            prediction_data_column_name=prediction_data_column_name,
+            ground_truth_column_name=ground_truth_column_name,
+            metrics_mapping=metrics_mapping,
+            metrics=metrics,
+            type_to_kwargs=type_to_kwargs,
+        )
+
+        self._validate()
+
+    def _validate(self):
+        supported_list = [isinstance(metric, CodeMetric) for metric in self.metrics]
+        if not all(supported_list):
+            raise Exception(f"{self.__class__.__name__} supports only {CodeMetric.__class__.__name__} type of metrics")
+
+    def calculate_metrics(self):
+        LOGGER.info(f"Calculating code metrics : {[metric.name for metric in self.metrics]}")
+        metrics_dict = {"artifacts": {}, "metrics": {}}
+        metric_results_futures = {}
+        with tqdm.tqdm(total=len(self.metrics)) as progress_bar:
+            with ThreadPoolExecutor(thread_name_prefix="code_metrics") as thread_pool:
+                for metric in self.metrics:
+                    metric_values = []
+                    metric_results_futures.update({metric.name: thread_pool.submit(
+                        self._calculate_metric, metric, self.test_data.to_dict('records'),
+                        self.prediction_data.to_dict('records') if self.prediction_data is not None else None
+                    )})
+
+                for metric_name, metric_result_future in metric_results_futures.items():
+                    try:
+                        metric_result = metric_result_future.result()
+                        metrics_dict["artifacts"].update(metric_result["artifacts"])
+                        if "metrics" in metric_result.keys() and metric_result["metrics"] is not None:
+                            metrics_dict["metrics"].update(metric_result["metrics"])
```

Comment:
Shall we do a metric name uniqueness check early on to avoid metrics overwriting each other?